src/FastSerialization/GrowableArray.cs

```diff
@@ -18,7 +37,7 @@ namespace System.Collections.Generic
 #if GROWABLEARRAY_PUBLIC
     public
 #endif
-    struct GrowableArray<T>
+    struct GrowableArray<T> : IMutableList<T>
```

Comment:
⚠️  This one is a possible problem. If `GrowableArray<T>` gets boxed, it could get corrupted. Implementing an interface makes it much easier for this to occur.

Reply:
For the specific locations where `SegmentedList` was introduced, I don't believe a performance problem will occur. In addition, the segmentation algorithm used by this type is quite fast.